torch/_subclasses/meta_utils.py

```diff
@@ -1448,7 +1455,7 @@ def is_c_of_r(complex_dtype, real_dtype):
                                 device="meta",
                             )
                         )
-                        if self.copy_data:
+                        if self.copy_data and not has_symbolic_size(t.size):
```

Comment:
Is @pianpwk the right person to confirm this is valid?

Reply:
definitely not lol

Reply:
could you make a separate PR with the changes in this file, and include a test case and have Ed take a look?